### DISCUSSION TITLE
Power Consumption Comparison: ATtiny804 vs ATtiny404?

### DISCUSSION THREAD

**Discussion by krishank652** [SpenceKonde/megaTinyCore#1182] [Q&A]
Hello Community,

I am working with the ATtiny804 and have optimized it for power savings, but I still need to reduce its power consumption further. If I use the same code on the ATtiny404, will it consume less power than the ATtiny804, or will the power consumption be the same with the same code?

Regards,
Krishan

**Comment by MX682X** [comment]
Looking at the datasheets, the 404 has actually a higher consuption. Active CPU Consuption at 5V20MHz: 9mA(404) and 7.8mA(804) typical. The power consumtpion of peripherals doesn't change.

**Comment by SpenceKonde** [comment]
```
ATtiny 202/204/402/404/406

Active Active power consumption 
Conditions:                             | VDD = 5  | VDD = 3 | VDD = 2 |
CLK_CPU = 20 MHz (OSC20M) VDD           | 9.0 mA
CLK_CPU = 10 MHz (OSC20M div2)          | 4.8 mA   | 2.7 mA
CLK_CPU = 5 MHz (OSC20M div4)           | 2.8 mA   | 1.6 mA  | 1.0 mA
CLK_CPU = 32.768 kHz (OSCULP32K)        | 18 µA    | 10 µA   | 7 µA
Idle Idle power consumption 
CLK_CPU = 20 MHz (OSC20M)               | 2.8 mA   | -       | -
CLK_CPU = 10 MHz (OSC20M div2)          | 1.7 mA   | 0.9 mA  | -       | -
CLK_CPU = 5 MHz (OSC20M div4)           | 1.2 mA   | 0.6 uA  | 0.4 mA
CLK_CPU = 32.768 kHz (OSCULP32K)        | 8.0 µA   | 4.0 µA  | 1.7  µA



ATtiny 804/806/807/1604/1606/1607
Conditions:                             | VDD = 5  | VDD = 3 | VDD = 2 |
CLK_CPU = 20 MHz (OSC20M)               | 7.8 mA   | -       | -       |
CLK_CPU = 10 MHz (OSC20M div2)          | 4.3 mA   | 2.4 mA  | -       |
CLK_CPU = 5 MHz (OSC20M div4)           | 2.4 mA   | 1.9 mA  | 0.9 mA  |
CLK_CPU = 32.768 kHz (OSCULP32K)        | 16 µA    | 9.2 µA  | 6.3 µA
Idle Idle power consumption 
CLK_CPU = 20 MHz (OSC20M)               | 2.1 mA   | -       | -
CLK_CPU = 10 MHz (OSC20M div2)          | 1.3 mA   | 0.7 mA  | -  
CLK_CPU = 5 MHz (OSC20M div4)           | 0.9 mA   | 0.5 mA  | 0.3 mA
CLK_CPU = 32.768 kHz (OSCULP32K)      | 4.3 µA   | 2.1 µA  | 1.4 µA
```

As MX682X said, you'd be *worse off* for power consumption on the 404. 


```
For comparison, a 1614/1616/1617
Active power consumption
CLK_CPU = 20 MHz (OSC20M)        |  10.2 mA |  -      |  - 
CLK_CPU = 10 MHz (OSC20M div2)   |  5.5 mA  |  3.1 mA |  -
CLK_CPU = 5 MHz(OSC20M div4)     |  3.2 mA  |  1.8 mA |  1.2 mA
CLK_CPU = 32.768 kHz(OSCULP32K)  |  13.5 µA |  7.5 µA |  5.0 µA
Idle power consumption
CLK_CPU = 20 MHz (OSC20M)        |  4.3 mA |  -      |  - 
CLK_CPU = 10 MHz (OSC20M div2)   |  2.2 mA |  1.1 mA |  - 
CLK_CPU = 5 MHz (OSC20M div4)    |  1.1 mA |  0.6 0.9 mA |  0.4 mA
CLK_CPU = 32.768 kHz (OSCULP32K) |  8.2 ua |  4.2 µA |  2.6 µA
Active power consumption
CLK_CPU = 20 MHz (OSC20M) 		 |  9.1 mA |  -      |  - 
CLK_CPU = 10 MHz (OSC20M div2)   |  4.9 mA |  2.8 mA |  - 
CLK_CPU = 5 MHz  (OSC20M div4)   |  2.8 mA |  1.6 mA |  1.1 mA
CLK_CPU = 32.768 kHz (OSCULP32K) |  18 µA  |  10 µA  |  7.2 µA
Idle power  consumption
CLK_CPU = 20 MHz (OSC20M)        |  2.4 mA |  -      |  - 
CLK_CPU = 10 MHz(OSC20M div2)    |  1.4 mA |  0.8 mA |  - 
CLK_CPU = 5 MHz(OSC20M div4)     |  1.0 mA |  0.6 mA |  0.4 mA
CLK_CPU = 32.768 kHz(OSCULP32K)  |  4.8 µA |  2.4 µA |  1.5 µA


14/20 pin AVR DD:
All values at 3.0 volts. Notice that this newer chip usually beats the modern tinyAVRs.

Active power consumption
4.7 mA OSCHF = 24 MHz
1.1 mA OSCHF = 4 MHz
4.7 μA OSC32K = 32.768 kHz
4.4 mA EXTCLK = 24 MHz
910 μA EXTCLK = 4 MHz
760 μA XTALHF = 4 MHz,
XOSCHFCTRLA.FRQRANGE = 0x0
3.32 mA XTALHF = 24 MHz,
XOSCHFCTRLA.FRQRANGE = 0x2
8.2 μA XOSC32K = 32.768 kHz,
XOSC32KCTRLA.LPMODE = 0
6.3 μA XOSC32K = 32.768 kHz,
XOSC32KCTRLA.LPMODE = 1
Idle power consumption
2.3 mA OSCHF = 24 MHz
630 μA OSCHF = 4 MHz
3.1 μA OSC32K = 32.768 kHz
2.0 mA EXTCLK = 24 MHz
480 μA EXTCLK = 4 MHz
393 μA XTALHF = 4 MHz,
XOSCHFCTRLA.FRQRANGE = 0x0
1.37 mA XTALHF = 24 MHz,
XOSCHFCTRLA.FRQRANGE = 0x2
6.8 μA XOSC32K = 32.768 kHz,
XOSC32KCTRLA.LPMODE = 0
4.8 μA XOSC32K = 32.768 kHz,
XOSC32KCTRLA.LPMODE = 1

Minimum power consumption in different sleep modes
650 nA Power-Down or Standby mode,
VREGCTRL.PMODE = 0 (AUTO)
160 μA Power-Down mode,
VREGCTRL.PMODE = 1 (FULL)
0.9 μA
Power-Down mode,
VREGCTRL.PMODE = 0 (AUTO),
VREGCTRL.HTLLEN = 1
IRST Reset power consumption 170 μA RESET line pulled low
```

Anyway, one last **SUPER IMPORTANT THING**: Are you concerned about sleep mode power consumption being too high, or active/idle mode? 

**IF IT IS SLEEP MODE YOU ARE HAVING ISSUES WITH** you most likely have left pins floating. That is a big no-no now. 

In powerdown and standby, you *CANNOT* have any floating pins with the input buffer still enabled; at least not if you want to see decent sleep mode power consumption. 

Despite what the datasheets claim (that setting the pins output and low, or input with pullup on will also prevent this excessive drain), users have consistently reported that the only turning off the digital input buffer on all floating pins ( ie, for PIN_xn, `uint8_t temp = PORTx.PINnCTRL; temp &= 0xF8; temp |= 0x04; PORTx.PINnCTRL = temp` in the general case. We also provide a wrapper around PINnCTRL called  pinConfigure (see the digital I/O section of the core documentations for more info)). 

Note that for power save and sleep purposes, most people are just going to set the unused pins' PINnCTRL to 0x04. Disable the input buffer of  any pins that would be floating while in sleep prior to sleeping the chip. Only pins that are floating (not connected to anything, or connected to something that has the pin tristated).

This effect is claimed by Microchip to have always been like that, even on classic AVRs and that nothing has changed about that, which anyone who has used sleep on both classes of parts knows ain't so.

Personally, I am effectively certain that the cause of the increased susceptibility to this sort of power drain is that it's the price we pay for the improved interrupt capabilities - the pins are picking up noise, causing the buffers toflip back and forth rapidly, using a bit of power each time.
